File: translations/codenet/uncommented/Go2C/granite/Code/s054878611.c

#include <stdio.h>

int max(int x, int y) {
        if (x > y) {
                return x;
        }
        return y;
}

int solution(int n, int a[]) {
        int memo[100002] = {0, };
        for (int i = 0; i < n; i++) {
                memo[a[i]]++;
                memo[a[i] + 1]++;
                memo[a[i] + 2]++;
        }
        int out = 0;
        for (int i = 0; i < 100002; i++) {
                if (memo[i] > 0) {
                        out = max(out, memo[i]);
                }
        }
        return out;
}

int main() {
        int n;
        scanf("%d", &n);
        int a[n];
        for (int i = 0; i < n; i++) {
                scanf("%d", &a[i]);
        }
        printf("%d\n", solution(n, a));
        return 0;
}

File: translations/codenet/uncommented/Go2J/starcoder/Code/s054878611.java
package main;

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }
        System.out.println(solution(n, a));
    }

    public static int solution(int n, int[] a) {
        int[] memo = new int[100000 + 2];
        for (int num : a) {
            memo[num]++;
            memo[num + 1]++;
            memo[num + 2]++;
        }
        int out = 0;
        for (int num : memo) {
            if (num > 0) {
                out = Math.max(out, num);
            }
        }
        return out;
    }
}

